rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,761 @@
+                $ref: '#/components/schemas/GetTableResponse'
+        '401':
+          description: Unauthorized
+        # Using 412, `Precondition Failed`, instead of 404, as 404 makes monitoring response codes from ELBs
+        #  very difficult - Hard to tell if clients or servers are misconfigured and calling non-existent routes
+        #  or missing routes versus expected error cases such as NoSuchTableException (expected meaning that
+        #  a person who is on call shouldn't be paged for this but 404 they might need to be).
+        '412':
+          description: NoSuchTableException
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchTableError'
+    put:
+      tags:
+        - Catalog API
+      summary: Commit an in progress create (or replace) table transaction
+      operationId: commitTable
+      description: Commit a pending create (or replace) table transaction, e.g. for doCommit.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CommitTableRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CommitTableResponse'
+        '401':
+          description: Unauthorized / Invalid Credentials
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergResponseObject'
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a table from the catalog, optionally purging the underlying data
+      operationId: dropTable
+      description: Remove a table from the catalog, optionally dropping the underlying data
+      parameters:
+        - name: purge
+          in: query
+          required: false
+          schema:
+            type: boolean
+            default: false
+      responses:
+        '200':
+          description: OK
+          content:
+            'application/json':
+              schema:
+                type: boolean
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a table with a given identifier exists
+      operationId: tableExists
+      description:
+        Check if a table exists within a given namespace. Returns the standard response with `true` when found. Will return a TableNotFound error if not present. Can change to returning a 200 with a body of `false` if not found, but that does add more wok on the client.
+      parameters:
+        - name: namespace
+          in: path
+          required: true
+          schema:
+            type: string
+        - name: table
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+        '412':
+          description: Table Not Found
+  /v1/tables:
```

Comment:
I think this should be `/v1/namespaces/{namespace}/tables`?

Reply:
I originally had it that way, but the one use case that I wanted to be able to support was listing all tables, regardless of namespace. 

That's why for this path, I have `namespace` as a query parameter. I don't love the inconsistency though so would appreciate any thoughts on how to support both while being as consistent as possible.

Reply:
I agree with Jack. For consistency let's use `/v1/namespaces/{namespace}/tables`. We can consider adding a route for all tables later, but there is no corresponding method in the `Catlaog` API so I'm not sure that we need it. Let's concentrate on things in the exsting API for now to keep this small.

Reply:
Oh you're right. I was thinking of the `listNamespaces()` call that lists all top level namespaces.

I will update this then as I much prefer to keep it consistent. My bad!

Reply:
Updated.